diamond-api/src/application.rs

```diff
@@ -2,11 +2,24 @@ use crate::find::*;
 use crate::opts::*;
 use crate::render::*;
 use actix_web::web::{ServiceConfig, resource};
+use std::path::PathBuf;
 
-pub fn app_config(args: Args) -> impl Fn(&mut ServiceConfig) {
+#[derive(Clone)]
+pub struct Context {
+    pub args: Args,
+    pub walker: Walker,
+}
+
+#[derive(Clone)]
+pub enum Walker {
+    File(PathBuf),
+    Const(Vec<RenderPoint>),
```

Comment:
I do not understand why do we need this?

Reply:
This allows to decouple `render_handler` from `walk`, and cover `render_handler` with unit-tests without needing the real files to for walker to walk through.

Reply:
Since it is required only for unit-tests(like mock), it tests only this branch, which is never used in the program. Moreover it overcomplicates the code.

BTW, I have never seen this approach in rust.

@andy128k, @loony-bean, @Deitrl your thoughts?

Reply:
Looks like ol-good polymorphism is needed. 
```rust
trait Walker { fn walk(); }

struct DirectoryWalker(PathBuf);
impl Walker for DirectoryWalker { ... }

struct ConstWalker(Vec<RenderPoint>);
impl Walker for ConstWalker { ... }

struct Context {
   walker: Box<dyn Walker>,
}
```

Reply:
Thanks for comments, guys! I will address them today in the evening.

Reply:
> it is required only for unit-tests

I'd say opportunity for pain-less unit-testing is a significant benefit of de-coupling the implementation. In something like relay mode `Proxy` mode can be added here (will take 2 lines of code), and then easily tested.

I'm not saying this is the best possible solution, but it is really simple and clean, doesn't create redundant entries/structs/etc, and takes literally extra 10 lines of code.

> it overcomplicates the code

5 lines of code for a struct, 5 more for `walker(...) -> impl ...`

> I have never seen this approach in rust

It doesn't mean the approach is bad, does it? :)

> ol-good polymorphism

Tried it first, compiler was complaining about resolving reference lifetimes.
Then I don't think `Box` plays nice with Actix `.data()`, but I didn't check that.

Your suggestion introduces 5 new entities in total vs. 2 (`walker()` and enum).

Reply:
My idea was to provide simple enum as part as application config.
Then based on such enum, `walker()` returns the lambda that does the job.
Then result of `walker()` are used in `render_handler`.

This is the simplest impl I managed to come up with after 3 different approaches (polymorphism, method extraction, factory method). It allows to provide unit-test in clean and simple manner for the whole handler, I don't see why you don't like it.

Reply:
`Walker` is a pre-cursor to more generic "storage interface". That's why we are so overcautious.

Reply:
I am ok with current solution. Just not to block CSV feature.